test/sorting.jl

```diff
@@ -11,7 +11,7 @@ using .Main.OffsetArrays
 
 @testset "Order" begin
     @test Forward == ForwardOrdering()
-    @test ReverseOrdering(Forward) == ReverseOrdering() == Reverse
+    @test Reverse == ReverseOrdering() == ReverseOrdering(Forward) == ReverseOrdering{ForwardOrdering}()
 end
```

Comment:
```suggestion
    let T = ReverseOrdering{ReverseOrdering{ForwardOrdering}}
        @test T() isa T
    end
end
```
Also test the recursive case